packages/gatsby-source-drupal/src/utils.js

```diff
@@ -262,10 +261,12 @@ ${JSON.stringify(nodeToUpdate, null, 4)}
 
     const nodeFieldName = `${newNode.internal.type}___NODE`
     removedReferencedNodes.forEach(referencedNode => {
-      referencedNode.relationships[nodeFieldName] =
-        referencedNode.relationships[nodeFieldName].filter(
-          id => id !== newNode.id
-        )
+      if (referencedNode.relationships[nodeFieldName]) {
```

Comment:
We're sure that relationships always exists? e.g. could go further like:

```suggestion
      if (referencedNode.relationships && referencedNode.relationships[nodeFieldName]) {
```